lib/cosmos/tools/cmd_tlm_server/interface_thread.rb

```diff
@@ -58,9 +59,13 @@ def start
         begin
           Logger.info "Starting packet reading for #{@interface.name}"
           while true
+            break if @cancel_thread
             unless @interface.connected?
               begin
-                connect()
+                @mutex.synchronize do
+                  break if @cancel_thread
```

Comment:
Yeah - Its not doing exactly what I intended.  It is only breaking out of the mutex block instead of the entire thread.  That is preventing the connect(), but not exactly what I intended. I'll update.